app.ts

```diff
@@ -0,0 +1,426 @@
+					rangeArray = idRange.split('-');
+					$('.results-box').remove();
+					startID = parseInt(rangeArray[0]);
+					endID = parseInt(rangeArray[1]);
+					if (!isNaN(endID) && Number.isInteger(endID)) {
+						let maxRecords = this.recordsPerPage();
+						this.currentPage = Math.floor((endID + 1) / (maxRecords + 1));
+						pageEnd = Math.ceil(this.currentPage / 10) * 10;
+						pageStart = pageEnd - 9;
+						if (endID >= count) {
+							startID = ((this.currentPage - 1) * maxRecords) + 1;
+							endID = (count - 1);
+							this.paginationStart = pageStart;
+							this.paginationEnd = pageEnd;
+						}
+					} else {
+						throw new Error("Please provide a valid integer.");
+					}
+				})
+				.then(() => {
+					return this.pageNumbers(pageStart, pageEnd);
+				})
+				.then(() => {
+					return this.showRecords(startID, endID);
+				})
+				.then(() => {
+					$('.results-select').prop('disabled', false);
+				})
+				.catch(error => {
+					alert("An error occurred while trying to search. Please try again.");
+					console.error("Failed when clicking on the results: ", error);
+					throw error;
+				});
+		});
+	}
+
+	/** Will calculate the amount records to be shown according to the screen height. */
+	adjustDisplayedRecords(): Promise<void> {
+		let pageStart: number;
+		let pageEnd: number;
+		return this.getRecordCount()
+			.then(count => {
+				let maxRecords = this.recordsPerPage();
+				let inputNumber = this.input();
+				let newToID = this.currentToID === 0 ? this.currentToID + 1 : this.currentToID;
+				let newMaxRecords = maxRecords === 0 ? maxRecords + 1 : maxRecords;
+				if (inputNumber === -1) {
+					let newCurrentPage = Math.ceil(this.currentFromID / newMaxRecords);
+					if (newCurrentPage === 0) {
+						this.currentFromID = 0;
+						newCurrentPage = 1;
+					}
+					this.currentToID = this.currentFromID + maxRecords;
+					this.currentPage = newCurrentPage;
+					let originalID = (this.currentPage - 1) * (maxRecords + 1);
+					this.difference = originalID - this.currentFromID;
+				} else {
+					if (this.currentToID >= count) {
+						this.currentToID = (count - 1);
+					}
+					let newCurrentPage = Math.ceil(inputNumber / maxRecords);
+					this.currentToID = newCurrentPage * maxRecords;
+					this.currentPage = newCurrentPage;
+					this.currentFromID = (this.currentPage - 1) * maxRecords + 1;
+				}
+				pageEnd = Math.ceil(Math.floor(newToID / newMaxRecords) / 10) * 10;
+				pageEnd = pageEnd === 0 ? 10 : pageEnd;
+				pageStart = pageEnd - 9;
+				this.paginationStart = pageStart;
+				this.paginationEnd = pageEnd;
+				$("#records-table-body").empty();
+				return this.showRecords(this.currentFromID, this.currentToID);
+			})
+			.then(() => {
+				return this.pageNumbers(pageStart, pageEnd);
+			})
+			.catch(error => {
+				console.error("Failed when adjusting the records: ", error);
+				alert("An error occurred while trying to adjust the records. Please try again.");
+				throw error;
```

Comment:
???

So you handle the error... and then you throw it anyway meaning that you have to handle it again later?

Please do not handle the error here, as who ever is calling `adjustDisplayedRecords` needs to know that we failed to `adjustDisplayedRecords` and they need to react accordingly.